Reply to

petermarathas opened the pull request:
Application templates now contain a configurable reply-to address.

dfaulken commented on app/mailers/jobapps_mailer.rb:
```diff
@@ -8,22 +8,32 @@ class JobappsMailer < ActionMailer::Base
     @application_record = application_record
     @user = application_record.user
+    position = application_record.position
+    reply_to = position.application_template.present? ?
```
I agree with CodeClimate - better to avoid ternary statements

dfaulken commented on spec/factories/application_templates.rb:
```diff
@@ -2,6 +2,7 @@
   factory :application_template do
     position
     active true
+    email 'steve@sharklazers.com'
```
Do application templates require an email to be valid?

petermarathas replied:
Nope. Does that mean this should be removed?

dfaulken replied:
Yes. Factories should only contain the attributes which are necessary in order to make a valid instance.

dfaulken commented:
Does this close an issue?

petermarathas commented:
Closes #115

petermarathas commented:
@dfaulken I added <a href="https://github.com/umts/jobapps/pull/239/files#diff-1717d3e3835cc67fc19f0620ff5bc59aR109">this</a> and <a href="https://github.com/umts/jobapps/pull/239/files#diff-fcfc74e9d1b177a47d76c4345065d388R39">these</a> methods to make it easier to get application template and position. I was getting annoyed by how much I had to write each time I wanted an interview's original template and the same for record. 

Also I added <a href="https://github.com/umts/jobapps/pull/239/files#diff-5ba8e59a70da381cbe96315ecc39389dR11">this check</a> because there were some tests in which a record's position wouldn't have a template. I'm not sure if there would ever be a real occurrence of an application record not having an application template.

petermarathas commented:
Nevermind, code climate didn't like those methods. Still though, would there ever be a time where an application record would exist but its position would not have an application template?

dfaulken commented:
Sure - if an application was made in the past and then the position was no longer offered, we might delete the template.

dfaulken commented on app/mailers/jobapps_mailer.rb:
```diff
@@ -9,11 +9,11 @@ def application_denial(application_record)
-    else
-      reply_to = ''
-    end
+    reply_to = if template
```
Could also do

```ruby
reply_to = template.try(:email)
```

unless the mailer has a problem with a nil address rather than a blank string.

dfaulken commented on coverage/.last_run.json:
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 100.0
+    "covered_percent": 99.69
```
Oops - looks like both conditional branches aren't tested.

dfaulken commented on app/mailers/jobapps_mailer.rb:
```diff
@@ -31,7 +36,12 @@ def interview_confirmation(interview)
     @user = interview.user
-    reply_to = interview.application_record.position.application_template.email
+    template = interview.application_record.position.application_template
+    reply_to = if template
```
Will the mailer accept a `nil` reply_to value? If so, you could just do

```ruby
reply_to = template.try :email
```

dfaulken commented:
I'll take care of finishing this PR up @petermarathas. Thanks for getting it this far

dfaulken commented:
Merging - I'm okay with those CodeClimate issues.
